src/main/java/sirius/biz/analytics/charts/explorer/ChartFactory.java

```diff
@@ -0,0 +1,265 @@
+    private static final String OUTPUT_REFERENCE = "reference";
+    private static final String OUTPUT_IDENTIFIER = "identifier";
+
+    @Part
+    private GlobalContext globalContext;
+
+    private ChartObjectResolver<O> resolver;
+    private List<ChartFactory<O>> referencedProviders;
+
+    /**
+     * Specifies the resolver used to suggest and resolve objects for which this chart is provided.
+     *
+     * @return the resolver to use or <tt>null</tt> if this chart does not depend on a data object
+     */
+    @Nullable
+    protected abstract Class<? extends ChartObjectResolver<O>> getResolver();
+
+    protected ChartObjectResolver<O> resolver() {
+        if (resolver == null && getResolver() != null) {
+            resolver = globalContext.getPartByType(ChartObjectResolver.class, getResolver());
+        }
+
+        return resolver;
+    }
+
+    /**
+     * Determines if the chart uses a resolver or not.
+     *
+     * @return <tt>true</tt> if a resolver is used, <tt>false</tt> otherwise
+     */
+    public boolean hasResolver() {
+        return getResolver() != null;
+    }
+
+    /**
+     * Determines if the chart is accessible by the current user.
+     * <p>
+     * The default implementation evaluates all {@link sirius.web.security.Permission} annotations present on
+     * the class-level.
+     *
+     * @return <tt>true</tt> if the current user can access a chart, <tt>false</tt> otherwise
+     */
+    public boolean isAccessibleToCurrentUser() {
+        return UserContext.getCurrentUser()
+                          .hasPermissions(Permissions.computePermissionsFromAnnotations(getClass())
+                                                     .toArray(EMPTY_STRING_ARRAY));
+    }
+
+    /**
+     * Determines the name of the category to put this chart in.
+     * <p>
+     * Most probably this should use a constant (like one in {@link sirius.biz.jobs.StandardCategories} for SIRIUS or
+     * foundation classes).
+     *
+     * @return the name of the category to which this chart belongs
+     */
+    public abstract String getCategory();
+
+    /**
+     * Returns the CSS class which specifies the icon to show.
+     *
+     * @return the CSS class (i.e. fontawesome) to determine the icon to use
+     */
+    public abstract String getIcon();
+
+    /**
+     * Returns the label to show in the overview UI.
+     *
+     * @return the label to show for this chart. By default, this will fetch the property <tt>ClassName.label</tt>
+     */
+    public String getLabel() {
+        return NLS.get(getClass().getSimpleName() + ".label");
+    }
+
+    /**
+     * Returns the description to show in the overview UI.
+     *
+     * @return the description to show for this chart. By default, this will fetch the property <tt>ClassName.description</tt>
+     */
+    public String getDescription() {
+        return NLS.getIfExists(getClass().getSimpleName() + ".description", null).orElse("");
+    }
```

Comment:
Could be discussed if these methods should be called `build...` instead of `get...`, but in my opinion simple NLS lookups or string constructions are on the edge of counting as a typical getter

Reply:
as the JobFactory does pretty much the same I'd keep it this way (whil still agreeing that getters should be dead simple)...